packages/fuel-gauge/src/transaction-summary.test.ts

```diff
@@ -314,6 +314,131 @@ describe('TransactionSummary', () => {
       });
     });
 
+    it('should ensure transfer operation is assembled (UNLOCKING PREDICATE)', async () => {
+      const amountInPredicate = 100_000_000;
+      const amountToUnlockFromPredicate = 1_000_000;
+      const amountInFeePayeeUserWallet = 10_000;
+
+      using launched = await launchTestNode();
+
+      const {
+        provider,
+        wallets: [genesisWallet],
+      } = launched;
+
+      const baseAssetId = provider.getBaseAssetId();
+
+      // Make predicate and fund it
+      const predicate = new PredicateTrue({ provider });
+      const funding = await genesisWallet.transfer(
+        predicate.address,
+        amountInPredicate,
+        baseAssetId
+      );
+      await funding.waitForResult();
+
+      // Add some funds to the payee wallet
+      const feePayeeWallet = Wallet.generate({ provider });
+      const fundingFeePayee = await genesisWallet.transfer(
+        feePayeeWallet.address,
+        amountInFeePayeeUserWallet,
+        baseAssetId
+      );
+      await fundingFeePayee.waitForResult();
+
+      // Recipient wallet
+      const recipient = Wallet.generate({ provider });
+
+      // Unlocking funds from predicate request
+      const request = new ScriptTransactionRequest();
+      const predicateResources = await predicate.getResourcesToSpend([
+        [amountToUnlockFromPredicate, baseAssetId],
+      ]);
+      request.addResources(predicateResources);
+      request.addCoinOutput(recipient.address, amountToUnlockFromPredicate, baseAssetId);
+
+      // Fund the request from the fee payee wallet
+      const txCosts = await feePayeeWallet.getTransactionCost(request);
+      request.gasLimit = txCosts.gasUsed;
+      request.maxFee = txCosts.maxFee;
+      await feePayeeWallet.fund(request, txCosts);
+
+      // Send the transaction
+      const tx = await feePayeeWallet.sendTransaction(request);
+      const { operations } = await tx.waitForResult();
+
+      validateTransferOperation({
+        operations,
+        sender: predicate.address,
+        fromType: AddressType.account,
+        toType: AddressType.account,
+        recipients: [
+          {
+            address: recipient.address,
+            quantities: [{ amount: amountToUnlockFromPredicate, assetId: baseAssetId }],
+          },
+        ],
+      });
+    });
+
+    it('should ensure transfer operation is assembled (TWO ACCOUNTS TO ANOTHER ACCOUNT)', async () => {
```

Comment:
This case - when transfering the same quantities from two different accounts, then the resultant sender will be one of the accounts. Once I change the amount so they're not equal, we get two distinct operations (one with sender from A, and one from B) - which is what I expected in this case.

Reply:
@petertonysmith94 Which one was not the expected case? Having just one sender?

Why this was not expected?

Reply:
I would expect there to be two senders as there are two senders...?

When I updated `amountFromB` to 11, I obtained what I expected - two distinct transfer options for both senders (A and B).

Reply:
@petertonysmith94 But how do you know that there were 2 senders?

In your example, you funded the TX with UTXOs from 2 users, and have added `CoinOutputs` with predefined values, but how do you know that all fetched UTXOs were spent? 

The first funding user might have a UTXO with the amount of 1 ETH, and the VM could have used just this one to pay for the whole TX, and the other UTXO(s), including the ones from the second user, were not spent at all.

There is only so much where this validation can go through. Trying to cover all the edge cases at some point will become just assumptions. 

We could use the amount on the `ChangeOutput` and the fee value to better determine who transfers to who. However, the VM only accepts one `ChangeOutput` for each `assetId`. So it will be available only for just one sender.

It means that in your example case, if the VM chooses to spend 1 ETH UTXO to pay the whole TX, from the user that does not have its address specified on the `ChangeOutput` (second user since it was the last one to add resources to the TX), the user will lose all its change.

It is very likely that what is happening [here](https://github.com/FuelLabs/fuels-ts/issues/2939#issuecomment-2291491338) is because of a similar reason. Whoever implemented this transfer decided to set the users to receive the change of the spent UTXOs. And since the `ChangeOuptut` is already set for someone, they had added another `CoinOutput` transferring to self, to not lose UTXOs.

Reply:
Thank you @Torres-ssf for the through explanation, it's helped me better understand what is happening and my assumptions have been wrong in this case.